### PR TITLE
fix `mergeimports` issue in docgen

### PR DESCRIPTION
**Pull request by sujaypatil96**
Merge PR #1308 first

This PR seeks to solve the problem that @noelmcloughlin was facing in PR #1283

### CODE REVIEW

**Comment by sujaypatil96** [comment]
@noelmcloughlin: if you agree with the changes in this PR, then we can merge this in and close out PR #1283

File: linkml/generators/docgen.py [hunk 1]
```diff
@@ -159,6 +158,7 @@ def __post_init__(self):
         if self.example_directory:
             self.example_runner = ExampleRunner(input_directory=Path(self.example_directory))
         super().__post_init__()
+        self.schemaview = SchemaView(self.schema, merge_imports=self.mergeimports)
```
Comment (noelmcloughlin) [inline linkml/generators/docgen.py:161]: I see `self.mergeimports=False` here but https://github.com/linkml/linkml/issues/1282 is not fixed.

Comment (noelmcloughlin) [reply linkml/generators/docgen.py:161]: Wait, let me check something.

Comment (sujaypatil96) [reply linkml/generators/docgen.py:161]: I think you might be right @noelmcloughlin, we might have to pass the value of `self.mergeimports` to `all_classes()`, `all_slots()`, etc. in docgen.py like you have in PR #1283

Comment (noelmcloughlin) [reply linkml/generators/docgen.py:161]: I will check. I'm having an inconvenient bug in my parent schema ;-( just trying to fix that and I can test for sure.

Comment (sujaypatil96) [reply linkml/generators/docgen.py:161]: Awesome, thank you. I've updated this PR.

**Review by cmungall — APPROVED** [review]

**Comment by noelmcloughlin** [comment]
https://github.com/linkml/linkml/issues/1311

**Pull request merged**